libs/cardano-ledger-binary/test-lib/Test/Cardano/Ledger/Binary/Twiddle.hs

```diff
@@ -177,14 +199,25 @@ instance Twiddle Term where
   twiddle (TFloat x) = twiddle x
   twiddle (TDouble x) = twiddle x
 
+-- | Helper function for decoding an `Encoding` into a CBOR `Term`
 encodingToTerm :: Version -> Encoding -> Term
 encodingToTerm version enc =
   case decodeFull version (serialize version enc) of
     Right t -> t
     Left err -> error $ show err
 
+-- | Helper function for converting an arbitrary value into a CBOR `Term`
 toTerm :: ToCBOR a => Version -> a -> Term
 toTerm version = encodingToTerm version . toCBOR
 
+-- | Wraps an arbitrary value into a `Twiddler`
 toTwiddler :: Twiddle a => a -> Gen (Twiddler a)
 toTwiddler x = Twiddler x <$> twiddle x
+
+-- | Function for testing the invariant of a `Twiddle` instance. For a correct
+-- implementation, this property should always hold.
+twiddleInvariantProp :: forall a. Twiddle a => Version -> a -> Gen Bool
```

Comment:
Since it is a property it sis best to return it as such, this will have better output upon failure
```suggestion
twiddleInvariantProp :: forall a. Twiddle a => Version -> a -> Gen Property
```